packages/plugin-compiler/src/main/kotlin/io/realm/compiler/IrUtils.kt

```diff
@@ -150,9 +150,42 @@ internal fun IrPluginContext.lookupConstructorInClass(
 }
 
 object SchemaCollector {
-    val properties = mutableMapOf<IrClass, MutableMap<String, Pair<String, IrProperty>>>()
+    val properties = mutableMapOf<IrClass, MutableMap<String, SchemaProperty>>()
 }
 
+// ------------------------------------------------------------------------------
+
+/**
+ * This matches RealmEnums.CollectionType.
+ */
+enum class CollectionType {
+    NONE,
+    LIST,
+    SET,
+    DICTIONARY
+}
+
+/**
+ * This matches RealmEnums.PropertyType.
+ */
+enum class PropertyType {
+    RLM_PROPERTY_TYPE_INT,
+    RLM_PROPERTY_TYPE_BOOL,
+    RLM_PROPERTY_TYPE_STRING,
+    RLM_PROPERTY_TYPE_OBJECT,
+    RLM_PROPERTY_TYPE_FLOAT,
+    RLM_PROPERTY_TYPE_DOUBLE
+}
+
+data class SchemaProperty(
+    val type: String,
```

Comment:
We could consider to just use the declaration type instead of `type` or maybe extract the _element type_ into this `type` to mimic cores way of representing a property by a _type_ and a (potentially NONE) collection type. 

But I guess exposing the public scheme will probably lead to some clean up of this.

Reply:
I will add a fixme and point to the schema issue as agreed.